python/cudf/cudf/core/udf/typing.py

```diff
@@ -37,9 +37,18 @@ class MaskedType(types.Type):
     def __init__(self, value):
         # MaskedType in Numba shall be parameterized
         # with a value type
-        if not isinstance(value, SUPPORTED_NUMBA_TYPES):
-            raise TypeError("value_type must be a numeric scalar type")
-        self.value_type = value
+        if isinstance(value, SUPPORTED_NUMBA_TYPES):
+            self.value_type = value
+        else:
+            # Unsupported Dtype. Numba tends to print out the type info
+            # for whatever operands and operation failed to type and then
+            # output its own error message. Putting the message in the repr
+            # then is one way of getting the true cause to the user
+            self.value_type = types.Dummy(
+                "\n\n\n Unsupported MaskedType. If you are seeing this, "
+                "you are likely trying to use a column of unsupported"
+                f" dtype in a UDF. Unsupported type: {value} \n\n\n"
```

Comment:
Minor nit: align the space at the end of the line.
```suggestion
                "you are likely trying to use a column of unsupported "
                f"dtype in a UDF. Unsupported type: {value} \n\n\n"
```